risc0/zkvm/src/prove/tests.rs

```diff
@@ -270,6 +270,34 @@ fn continuation() {
     }
 }
 
+fn run_dev_mode() {
+    use crate::receipt::InnerReceipt;
+    std::env::set_var("DEV_MODE", "1");
```

Comment:
I think this is dangerous. If this test runs in parallel, it could break other tests. @mothran Mentioned this which seems apropos: https://stackoverflow.com/questions/35858323/how-can-i-test-rust-methods-that-depend-on-environment-variables/67433684#67433684

Reply:
So... I looked into this and realized that any test that does proof generation or verification needs to use this. I looked at the implementation of that crate and it seems to use a mutex on the env var. I wonder if it would be easier to mark all proof tests and verification tests as `serial`. On the other hand, it would be nice to run these dev mode tests in isolation. I feel bad having to turn all proving tests serial because of this one test...